src/nvim/edit.c

```diff
@@ -609,9 +609,6 @@ static int insert_execute(VimState *state, int key)
   InsertState *s = (InsertState *)state;
   s->c = key;
 
-  // Don't want K_EVENT with cursorhold for the second key, e.g., after CTRL-V.
-  did_cursorhold = true;
```

Comment:
FWIW, I agree that this line should be removed, but we need to investigate the implications this has for insert mode. For example, we can easily stop this from affecting ctrl+v with this patch:


```diff
diff --git a/src/nvim/edit.c b/src/nvim/edit.c
index 208e419..46860a6 100644
--- a/src/nvim/edit.c
+++ b/src/nvim/edit.c
@@ -618,9 +618,6 @@ static int insert_execute(VimState *state, int key)
   InsertState *s = (InsertState *)state;
   s->c = key;

-  // Don't want K_EVENT with cursorhold for the second key, e.g., after CTRL-V.
-  did_cursorhold = true;
-
   if (p_hkmap && KeyTyped) {
     s->c = hkmap(s->c);  // Hebrew mode mapping
   }
@@ -729,6 +726,7 @@ static int insert_execute(VimState *state, int key)
   }

   if (s->c == Ctrl_V || s->c == Ctrl_Q) {
+    did_cursorhold = true;
     ins_ctrl_v();
     s->c = Ctrl_V;       // pretend CTRL-V is last typed character
     return 1;  // continue
```

but there are other multi-key commands in insert mode that can be affected, so a way to fix this is to add `did_cursorhold = true` statements to every other case.

A better(and possibly easier) way to fix the problem is to remove cursorhold triggering from os/input.c and move it into state.c, which could check `NvimState` structures for a flag that enables/disables cursorhold. This is better because it would let us remove all globals related to cursorhold triggering, and would automatically handle cases like the ctrl+v above which don't use `state_enter` to block for characters.